source/code/plugins/changetracking_lib.rb

```diff
@@ -130,39 +122,46 @@ def self.transform_and_wrap(inventoryXMLstr, host, time, force_send_run_interval
         services = removeDuplicateCollectionNames(services)
         fileInventories = removeDuplicateCollectionNames(fileInventories)
 
-        if (packages.size > 0 or services.size > 0 or fileInventories.size > 0)
-            timestamp = OMS::Common.format_time(time)
-            wrapper = {
-              "DataType"=>"CONFIG_CHANGE_BLOB",
-              "IPName"=>"changetracking",
-              "DataItems"=>[
-                {
-                    "Timestamp" => timestamp,
-                    "Computer" => host,
-                    "ConfigChangeType"=> "Software.Packages",
-                    "Collections"=> packages
-                },
-                {
-                    "Timestamp" => timestamp,
-                    "Computer" => host,
-                    "ConfigChangeType"=> "Daemons",
-                    "Collections"=> services
-                },
-                {
-                    "Timestamp" => timestamp,
-                    "Computer" => host,
-                    "ConfigChangeType"=> "Files",
-                    "Collections"=> fileInventories
+        return {"packages" => packages, "services" => services , "fileInventories" => fileInventories}
+    end
+
+    def self.wrap (time, host, inventory_hash)
+        @@log.debug "The keys in inventory_hash - #{inventory_hash.keys}"
+        @@log.debug "The keys in inventory_hash - #{inventory_hash.keys}"
+        wrapper = {}
+        # inventory_hash.key?("inventory_hash")
+        if ((inventory_hash.has_key?("packages") and inventory_hash["packages"].size > 0) or 
+            (inventory_hash.has_key?("services") and inventory_hash["services"].size > 0) or
+            (inventory_hash.has_key?("fileInventories") and inventory_hash["fileInventories"].size > 0))
+                timestamp = OMS::Common.format_time(time)
+                wrapper = {
+                    "DataType"=>"CONFIG_CHANGE_BLOB",
+                    "IPName"=>"changetracking",
+                    "DataItems"=>[
+                        {
+                            "Timestamp" => timestamp,
+                            "Computer" => host,
+                            "ConfigChangeType"=> "Software.Packages",
+                            "Collections"=> inventory_hash["packages"]
```

Comment:
If the key doesn't exist or the value is nil, shall we include it in the DataItems?